src/EditorFeatures/Core/EmbeddedLanguages/AbstractEmbeddedLanguageEditorFeaturesProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.Features.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.Features.EmbeddedLanguages.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Editor.EmbeddedLanguages
+{
+    /// <summary>
+    /// Abstract implementation of the C# and VB embedded language providers.
+    /// </summary>
+    internal abstract class AbstractEmbeddedLanguageEditorFeaturesProvider 
+        : AbstractEmbeddedLanguageFeaturesProvider, IEmbeddedLanguageEditorFeaturesProvider
+    {
+        new public ImmutableArray<IEmbeddedLanguageEditorFeatures> Languages { get; }
+         
+        protected AbstractEmbeddedLanguageEditorFeaturesProvider(EmbeddedLanguageInfo info) : base(info)
+        {
+            Languages = ImmutableArray.Create<IEmbeddedLanguageEditorFeatures>(
+                new RegexEmbeddedLanguageEditorFeatures(info));
```

Comment:
>RegexEmbeddedLanguageEditorFeatures [](start = 20, length = 35)

It would be good to comment why there is no fallback language in this layer (I think because it doesn't handle brace matching).

Reply:
Sure.  i can def add that comment.  (and that reason is correct.  the only thing the fallback language does is classify).

Reply:
Done.